### ISSUE TITLE
Proposal: The Null Ignition System

### ISSUE THREAD

**Issue by Jayhui-q** [hanabi/hanabi.github.io#644]
Avanderwalde, Rkass, and I in the past competition season (Null 5 Suits) decided to not use the positional clue system in the null conventions, and also to keep as much as possible from hyphen-ated conventions in a no variant environment. 

To do so, we needed to find a type of clue that did not interfere with any other type of clues -- clues that did not demand a certain already-defined response in order to synchronize the team.

There's a prime candidate for this: **an ignition.** As described in our current docs, "Nothing needs to be 'proved' to the player who got clued, which is unlike a Finesse, and unlike a Bluff." 

As such, I present to you the **Null Ignition System,** which is what we played with during the competition. There are three important details. First, null ignitions. Second, null saves, and Third, trash pull change. There are also a few null ignition extensions that aren't as essential. These are in order of their importance in the system.

### Null Ignition 
- Any clue that would ordinarily call for a double ignition (such as TDI, RDI, PDI) no longer calls for two plays from finesse. Instead, it is a _Null Ignition_, which ignites the **chop of a player**, who is promised a null playable on chop. This can be done in position or out of position.
- _Null Ignitions_ take precedence over other conventions that are not variant-specific.
- Null Ignitions behave with the same logic as ambiguous finesse, just like double ignition now does.
- For example:
  * In a four-player game of Null 5 Suits
  * Null 1 is on the stacks and Cathy has a known playable blue 1 that is clued with blue.
  * Alice clues blue to Cathy, focusing Cathy's blue 1.
  * While normally this would be a replay double ignition or a replay ejection, Bob knows instead this is a _Null Ingnition_. 
  * Bob looks at Cathy and Donald's hands and sees that neither of the players holds a null 2 on their chop. 
  * Bob knows that the _Null Ignition_ is on him, and plays chop. Null 2 correctly plays on the stacks. 
- Regarding edge case conflicts with trash push and trash chop move for the clue recipient, players make the same reads as they do with trash double ignitions in clue interpretation.
 
We found generally that null plays from chop were the most advantageous and most flexible location for the play to be called for, as if the card were not on chop, we could finesse/eject/discharge/charm it normally or just wait until the card gets to chop to get it played. Most of the time, there was no need to rush.

Here are a bunch of examples of null ignitions in our games

https://hanab.live/replay/499223#7 (with replay)
https://hanab.live/replay/499280#20 (with known trash)
https://hanab.live/replay/499280#36 (with known trash)
https://hanab.live/replay/493618#36 (with known trash)
https://hanab.live/replay/500373#21 (with replay - there's a misplay from a mistake on g2 save, but the u2 plays later)
https://hanab.live/replay/500446#29 (with known trash)
https://hanab.live/replay/500446#38 (with known trash)
https://hanab.live/replay/500479#29 (with known trash, we had turned off TPE, but it is not necessary)
https://hanab.live/replay/500479#38 (with known trash)
https://hanab.live/replay/501338#36 (with known trash)
https://hanab.live/replay/501393#5 (with replay)
https://hanab.live/replay/502323#28 (with known trash)
https://hanab.live/replay/503679#6 (with replay)

### Null Saves
- Normal 5 pulls, and 5 rank clues outside of low score phase that require more than two blind-plays from the same player to fulfill as a play clue are replaced with Null Saves (5 double pulls can still work normally)
- Null saves promise that a player has a Null 2 or a null critical on chop.
- 5 Chop Moves work as normal, taking higher precedence than null saves.
- This move can be performed by re-clueing with rank as well, when the clue would demand two-or-more blind-plays from the same player.

Null save examples:
https://hanab.live/replay/500373#6 (u5 save)
https://hanab.live/replay/495501#5 (Jerry knows u2 because it was promised as u2,5)
https://hanab.live/replay/500400#4 (example of where 5CM takes precedence)
https://hanab.live/replay/503566#6 (re-clue that demands two or more blind-plays)
https://hanab.live/replay/503738#6 (re-clue that demands two or more blind-plays)

### Trash Pull Change
This is much less important than the null ignition and null saves above, but we found that chop moved null cards can sometimes be pesky to work with. As such, we had all trash push clues given to chop moved players changed into trash pulls instead, pulling the chop moved card to the right.

Example:
https://hanab.live/replay/503738#42

### Nulll Ignition Extensions
We also experimented with null ignitions that moved slots away from chop depending on the specific card chosen for the null ignition. This was admittedly a lot to work with, and definitely requires very sensitive play. I definitely do not think these extensions are core to the above.

In our system, we had a precedence order for which cards to focus to do a null ignition. The player's highest precedence  would be clues focusing cards in the next player who is not the ignition recipient in order from left to right and color/rank. 
That is, color to slot 1 is highest precedence, then rank to slot 1, then color to slot 2, etc. The last level of precedence is any ignition clue that involves the ignition recipient's hand.

Players would play chop, then the rightmost chop moved card, then proceed leftwards in the hand to match the precedence.

Examples: 
https://hanab.live/replay/499223#25 (red would ordinarily get chop).
https://hanab.live/replay/502323#31 (blue to avanderwalde would ordinarily get chop)
https://hanab.live/replay/502385#35 (blue would get chop, 1 would get the chop moved card)
https://hanab.live/replay/503566#36 (yellow, 4, then 1s, in precedence)
______
tl;dr. We played null without positionals, keeping almost all of our typical clue tools, replacing double ignitions with a play from chop, and replacing 5 pulls & 5 rank play clues that require two or more blind-plays with a null save.

**Comment by pianoblook** [comment]
I definitely think positional Null can and should be retired, at least if the team is playing at max level!

A lot of this tracks pretty closely with the KiPiPi Null strats, laid out [here](https://docs.google.com/document/d/1idE_9n0y34B8-4BGJCtrAqF47pLi7s9sNOQ3UpDUWv8/edit#heading=h.ds5ky0b6ao6e). I believe the sj/Lib/ADrone used this system in the competition too. 


The big similarities are:
1) Turning off Ignitions in favor of _Kicks_ (chop blind-plays) - although only when it's a LPC, in our version.
2) Using 5 reclues as Null CMs

Key differences are:
1) We keep 5 pulls on, but also encourage 5 stalls in early game to give 5 Reclue-CM targets. IMO 5-pulls are actually extremely powerful in Null! I imagine the 'Trash CM Pull' system you created sort of serves a similar function though.
2) A lot of "Ignition" style clues (recluing trash) become Null CMs as well, whenever the player isn't otherwise Loaded. But most of the time yeah, that kind of clue does indeed get a Kick.


Big question, were you all playing it with HVRs or not?

**Comment by Jayhui-q** [comment]
Oh fascinating we didn't reference that document at all when creating ours -- very cool how we aligned through independent innovation. 

Assuming HVRs is Hard Variant Rules, then yes. Everything except for 5 saves/cm's were treated as play clues if the player had a known playable or known trash. Also, DDA and early game 5 stalls were off.

**Comment by Zamiell** [comment]
jerry, based on what is said in this thread, it looks like there isn't very much consensus on how the hyphen-ated group should play null (in the general non-competition sense)

maybe you can get together with piano's group and libster's group and come up with a unified system that is designed for casual-play and ultra-simple so that it can be used by people of all skill levels (for the purposes of supplanting the current positional clue system)
